tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -150,8 +150,17 @@ def run(self):
                                     connected=peers['peer%s.connected' % index]
                                 )
                             try:
+                                stored_peer_status = None
+                                # find peer detail using hostname
+                                ip  = socket.gethostbyname(
+                                    peers['peer%s.primary_hostname' % index]
+                                )
+                                node_id = etcd_utils.read(
+                                    "/indexes/ip/%s" % ip
+                                ).value
                                 stored_peer = NS.tendrl.objects.GlusterPeer(
-                                    peer_uuid=peers['peer%s.uuid' % index]
+                                    peer_uuid=peers['peer%s.uuid' % index],
```

Comment:
peer status always should check to form actual nodes peer directory, Because of a problem with the current approach is then node who is raising peer disconnected alert is only updating itself peer is a disconnect. But after few minutes it is deleted by TTL. So there is no meaning of marking the peer is down. So when no up there is no difference between old status and the new one. Because all other nodes have a value connected only for that particular peer. 

In this logic, I have modified the code like peer status fetching and updating happens actual nodes peer directory. So all nodes always look the same place.